lib/puppet/provider/package/pkgdmg.rb

```diff
@@ -68,8 +68,10 @@ def self.installpkg(source, name, orig_source)
   end
 
   def self.installpkgdmg(source, name)
-    http_proxy_host = Puppet::Util::HttpProxy.http_proxy_host
-    http_proxy_port = Puppet::Util::HttpProxy.http_proxy_port
+    unless Puppet::Util::HttpProxy.no_proxy?(source)
```

Comment:
Hi @branan ,

Ahh I see the tests you're referring to,  Ive added an additional test to account for the no_proxy variable.